pkg/reconciler/knativeserving/common/images.go

```diff
@@ -38,23 +38,23 @@ var (
 	containerNameVariable = "${NAME}"
 )
 
-func DeploymentTransform(instance *servingv1alpha1.KnativeServing, log *zap.SugaredLogger) mf.Transformer {
-	return func(u *unstructured.Unstructured) error {
-		// Update the deployment with the new registry and tag
-		if u.GetKind() == "Deployment" {
-			return updateDeployment(instance, u, log)
-		}
-		return nil
-	}
-}
-
+// ImageTransform updates image with a new registry and tag
 func ImageTransform(instance *servingv1alpha1.KnativeServing, log *zap.SugaredLogger) mf.Transformer {
 	return func(u *unstructured.Unstructured) error {
-		// Update the image with the new registry and tag
-		if u.GetAPIVersion() == "caching.internal.knative.dev/v1alpha1" && u.GetKind() == "Image" {
-			return updateCachingImage(instance, u)
+		switch u.GetKind() {
+		// TODO need to use PodSpecable duck type in order to remove duplicates of deployment, daemonSet
+		case "Deployment":
+			return updateDeployment(instance, u, log)
+		case "DaemonSet":
+			return updateDaemonSet(instance, u, log)
+		case "Image":
+			if u.GetAPIVersion() == "caching.internal.knative.dev/v1alpha1" {
+				return updateCachingImage(instance, u)
+			}
+			return nil
+		default:
+			return nil
 		}
-		return nil
```

Comment:
nit: i would lose the default case and replace the two returns in the switch with the old one at the end of the function.